validator/client/validator_propose.go

```diff
@@ -59,6 +73,12 @@ func (v *validator) ProposeBlock(ctx context.Context, slot uint64, pubKey [48]by
 	}
 	b.Signature = sig
 
+	db.SetProposedForEpoch(history, epoch)
```

Comment:
I mentioned it below but `db.SetProposedForEpoch(history, epoch)` and `v.db.SaveProposalHistory(pubKey[:], history)` should be called in one method

Reply:
Not sure how I feel about this change since `SetProposedForEpoch` is more of a helper function. Changing it to modify both the passed in object and the DB seems off.

Reply:
I've moved the functions to `validator/client` to prevent any confusion, it feels cleaner there.